test.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+set -e
+
+while getopts ":h" option; do
+   case $option in
+      h) # display Help
+         echo "This is a script for running the R-judge local"
+         echo
+         echo "Syntax: test.sh [-h] PATH_TO_EXERCISE_DIR PATH_TO_DUMP_DIR [PATH_TO_SOLUTION_FILE]"
+         echo "Param: \n\
+\tPATH_TO_EXERCISE_DIR    path to the exercise folder \n\
+\tPATH_TO_DUMP_DIR        path to the a folder where the exercise will be solved and stored \n\
+\t[PATH_TO_SOLUTION_FILE] optional path to the solution file default is set to \n\
+\t                        PATH_TO_EXERCISE_DIR/solution/solution.en.R"
+         echo "options:"
+         echo "h     Print this Help."
+         echo ""
+         echo
+         exit;
+   esac
+done
+
+path_to_exercise="$1"
+[ -z "$path_to_exercise" ] || [ ! -d "$path_to_exercise" ] \
+    && echo 'Provide a path to an exercise as first argument.' \
+    && exit 1
+
+test_dir="$2"
+[ -z "$test_dir" ] || [ ! -d "$test_dir" ]\
+    && echo 'Provide a path to a testing directory where the exercise will be solved and stored as second argument.' \
+    && exit 1
+
+path_to_solution=${3:-"$path_to_exercise/solution/solution.en.R"}
+[ ! -f "$path_to_solution" ] \
+    && echo "No solution found at $path_to_solution" \
+    && exit 1
+
+#find next available name in the test directory
```

Comment:
Is there a reason to keep previous directories? Why not just remove existing files in the directory?

Reply:
Because the directory is chosen by the tester i wasn't really comfortable with deleting the files in the folder but maybe it's a good idea to make an empty default directory in the judge git directory instead?

Reply:
You're right not to remove files to prevent accidents.

You could place a `.keep` file in the default empty directory (signalling git that the empty directory should be added), and check if that file exists before deleting the directory contents?

Reply:
I chose to use a separate `clean.sh` which cleans the current directory of files commonly created by the judge/exercises - this leaves the choice to the user.

Reply:
I will go for a combination of the two: 
1) An empty directory  test/ to store the results which I will make the default directory for testing.
2) A clean-script for the removal of the folders content.